packages/vm/vmcontext/migrations.go

```diff
@@ -0,0 +1,46 @@
+package vmcontext
+
+import (
+	"fmt"
+
+	"github.com/iotaledger/wasp/packages/kv"
+	"github.com/iotaledger/wasp/packages/vm/core/migrations"
+	"github.com/iotaledger/wasp/packages/vm/core/root"
+)
+
+func (vmctx *VMContext) runMigrations() {
+	if vmctx.task.AnchorOutput.StateIndex == 0 {
+		// initializing new chain -- set the schema to latest version
+		vmctx.callCore(root.Contract, func(s kv.KVStore) {
+			root.SetSchemaVersion(s, migrations.LatestSchemaVersion)
+		})
+		return
+	}
+
+	var schemaVersion uint32
+	vmctx.callCore(root.Contract, func(s kv.KVStore) {
+		schemaVersion = root.GetSchemaVersion(s)
+	})
+	if schemaVersion < migrations.BaseSchemaVersion {
+		panic(fmt.Sprintf("inconsistency: schema version %d should be >= %d", schemaVersion, migrations.BaseSchemaVersion))
+	}
+	if schemaVersion > migrations.LatestSchemaVersion {
+		panic(fmt.Sprintf("inconsistency: schema version %d should be <= %d", schemaVersion, migrations.LatestSchemaVersion))
+	}
+
+	for schemaVersion < migrations.LatestSchemaVersion {
+		migration := migrations.Migrations[schemaVersion-migrations.BaseSchemaVersion]
```

Comment:
shouldn't this be
```suggestion
		schemaVersion++
		migration := migrations.Migrations[schemaVersion]
```
?

if latest schema is 10 and current 3 this line is doing 3-10, that's weird.

Reply:
Let's say `base = 5`, `current = 7`, `latest = 10`, this would happen because there are 5 migrations in the list, the first one corresponding to schema version 5.

```
for schemaVersion < migrations.LatestSchemaVersion { // 7 < 10
  migration := migrations.Migrations[schemaVersion-migrations.BaseSchemaVersion] // 7-5 = 2
```

I think it's ok.